xbmc/utils/StringUtils.cpp

```diff
@@ -1048,29 +1048,29 @@ static wchar_t GetCollationWeight(const wchar_t& r)
 // returns negative if left < right, positive if left > right
 // and 0 if they are identical.
 // See also the equivalent StringUtils::AlphaNumericCollation() for UFT8 data
-int64_t StringUtils::AlphaNumericCompare(const wchar_t* left, const wchar_t* right)
+int64_t StringUtils::AlphaNumericCompare(std::wstring_view left, std::wstring_view right) noexcept
 {
-  const wchar_t *l = left;
-  const wchar_t *r = right;
-  const wchar_t *ld, *rd;
+  auto l = left.begin();
+  auto r = right.begin();
+  std::wstring_view::const_iterator ld, rd;
   wchar_t lc, rc;
   int64_t lnum, rnum;
   bool lsym, rsym;
-  while (*l != 0 && *r != 0)
+  while (l != left.end() && r != right.end())
   {
     // check if we have a numerical value
     if (*l >= L'0' && *l <= L'9' && *r >= L'0' && *r <= L'9')
     {
       ld = l;
       lnum = *ld++ - L'0';
-      while (*ld >= L'0' && *ld <= L'9' && ld < l + 15)
+      while (ld != left.end() && *ld >= L'0' && *ld <= L'9' && ld < l + 15)
```

Comment:
Causes a debug library assert "cannot seek string_view iterator after end", comparison of left=30 with right=60.

This should do the same and doesn't crash:

`while (ld != left.end() && *ld >= L'0' && *ld <= L'9' && ld - l < 15)`

But don't know if that takes care of all situations. The same preventively applied to line 1073 just in case.